File: jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXPagination.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.scene.control.Button;
import javafx.scene.layout.FlowPane;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.develnext.jphp.ext.javafx.classes.layout.UXFlowPane;
import org.develnext.jphp.ext.javafx.support.control.Pagination;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "UXPagination")
public class UXPagination extends UXFlowPane<Pagination> {
    interface WrappedInterface {
        @Property int total();
        @Property int pageSize();
        @Property int pageCount();
        @Property int maxPageCount();
        @Property int selectedPage();
        @Property String hintText();
        @Property boolean showTotal();

        @Property Button previousButton();
        @Property Button nextButton();
    }

    public UXPagination(Environment env, Pagination wrappedObject) {
        super(env, wrappedObject);
    }

    public UXPagination(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    @Signature
    public void __construct() {
        __wrappedObject = new Pagination();
    }
}

File: jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXPagination.php
<?php
namespace php\gui;

use php\gui\layout\UXFlowPane;

/**
 * Class UXPagination
 * @package php\gui
 */
class UXPagination extends UXFlowPane
{
    /**
     * @var int
     */
    public $total = 0;

    /**
     * @var int
     */
    public $pageSize = 20;

    /**
     * @readonly
     * @var int
     */
    public $pageCount;

    /**
     * @var int
     */
    public $maxPageCount = 7;

    /**
     * @var int
     */
    public $selectedPage = 0;

    /**
     * @var string
     */
    public $hintText = '';

    /**
     * @var bool
     */
    public $showTotal = false;

    /**
     * @readonly
     * @var UXButton
     */
    public $previousButton;

    /**
     * @readonly
     * @var UXButton
     */
    public $nextButton;

    public function __construct()
    {
    }
}